source/pick-from-possibly-undefined.d.ts

```diff
@@ -0,0 +1,32 @@
+/**
+`Pick` properties from type that may be undefined.
+
+@example:
+```
+import { PickFromPossiblyUndefined } from 'type-fest';
+
+type BillingDetails = {
+ taxId: string;
+ companyName: string;
+ address: string;
+ bankAccount: string;
+ ibanBankAccount: string;
+} | undefined;
+
+type CompanyBankAccounts = PickFromPossiblyUndefined<BillingDetails, 'bankAccount' | 'ibanBankAccount'>;
+
+const bankAccounts: CompanyBankAccounts = {
+ bankAccount: '123456789',
+ ibanBankAccount: '123456789',
+};
+
+await proceedPayment(bankAccounts);
+```
+
+@category Object
+ **/
+export type PickFromPossiblyUndefined<Type, Props extends keyof NonNullable<Type>> = Type extends undefined
```

Comment:
Why are you using `NonNullable`? We only want `undefined`, not `undefined` and `null`.

Reply:
How would you do that?